### PR TITLE
Add support for GoGo Board 6

### PR DESCRIPTION
**Pull request by kgmt0**
This adds a new daemon for GGB6 that can be selected at runtime using the new `init` primitive. Users are not required to use this primitive; if they run other commands without calling `init` first, the old daemon with be loaded automatically, so this PR should be fully backwards compatible.

Other than that, this PR also adds a new primitive `talk-to-servo-ports`. `talk-to-output-ports` do not affect servos on GGB6, so a new primitive is needed. I also upgraded `hid4java` and the included `libhidapi.so`. These seem to be necessary for GGB6 based on my testing.

### CODE REVIEW

File: src/HIDGogoExtension.java [hunk 1]
```diff
@@ -229,7 +224,7 @@ private String javaExecutablePath() {
   // can just spawn a process off from this one instead since everything is now
   // contained in a single jar.  But I don't want to make big changes like that
   // at the moment -Jeremy B January 2022
-  private void bootHIDDaemon() throws ExtensionException {
+  private void bootHIDDaemon(final boolean use_gogo6) throws ExtensionException {
```
Comment (LaCuneta) [inline src/HIDGogoExtension.java]: Nit: For Java code, we prefer `camelCase` for method parameters, so `useGoGo6` here.

**Review by LaCuneta — CHANGES_REQUESTED** [review]
Thank you for getting this working!  Code looks good, just one small item in the comments.

Is there any way to auto-detect which version of GoGo board is attached so that the user doesn't have to specify which with the new `init` prim?  I ask because needing to know ahead of time means a model made for GoGo 4 won't just run on GoGo 6 without a code change.  That could be an issue when sharing models with other GoGo users.

I don't have a GoGo board to test, but I did run a `compile` and `package`.  With the change to the HID library to not be a specified dependency in sbt, the NetLogo extension plugin didn't know to include it in the packaging, which will be needed as this is a bundled extension.  I think the easiest way to address this is adding a `netLogoPackageExtras` argument for it (and the Linux native lib, too).  [Here is the documentation for that feature](https://github.com/NetLogo/NetLogo-Extension-Plugin#including-extra-files-in-build-packaging-and-tests).  The R Extension makes use of it, if [an example to look at](https://github.com/NetLogo/R-Extension/blob/main/build.sbt#L13-L20) would be helpful.

Let me know if that all makes sense.

**Comment by kgmt0** [comment]
> Is there any way to auto-detect which version of GoGo board is attached so that the user doesn't have to specify which with the new init prim? I ask because needing to know ahead of time means a model made for GoGo 4 won't just run on GoGo 6 without a code change. That could be an issue when sharing models with other GoGo users.

It might be possible, but it would be a bit tricky. GoGo Boards all use the same product ID and vendor ID, so those don't work for detecting the board version. They do (or at least GGB6 does) have a field in the packets that gets sent to the controlling PC indicating the firmware version. However, I'm unable to confirm that firmware version doesn't get reset across board versions.

If the firmware version turns out to be usable, then we probably can launch the correct daemon automatically. However, it would require somewhat major changes. The daemon is responsible for reading the packet that contains the firmware version, so we have to start a daemon to get this information at all, and if necessary, shutdown it down and start a new one. This feels a bit error-prone to me, so I'm not sure it would be worth it.

> I don't have a GoGo board to test, but I did run a compile and package. With the change to the HID library to not be a specified dependency in sbt, the NetLogo extension plugin didn't know to include it in the packaging, which will be needed as this is a bundled extension. I think the easiest way to address this is adding a netLogoPackageExtras argument for it (and the Linux native lib, too). [Here is the documentation for that feature](https://github.com/NetLogo/NetLogo-Extension-Plugin#including-extra-files-in-build-packaging-and-tests). The R Extension makes use of it, if [an example to look at](https://github.com/NetLogo/R-Extension/blob/main/build.sbt#L13-L20) would be helpful.

I added `netLogoPackageExtras` to `build.sbt` as suggested and the jar seems to be included now. I'm not sure this is the right way to check, but I ran `packageZip` and confirmed that the jar is in the zip. The Linux native lib is already being included in the extension jar.

File: src/HIDGogoExtension.java [hunk 2]
```diff
@@ -242,9 +237,9 @@ private void bootHIDDaemon() throws ExtensionException {
     try {
       String classpath =
         new File(gogoExtensionPath + "gogo.jar").getCanonicalPath() + File.pathSeparator +
-        new File(gogoExtensionPath + "hid4java-0.7.0.jar").getCanonicalPath() + File.pathSeparator +
+        new File(gogoExtensionPath + "lib/hid4java-develop-SNAPSHOT.jar").getCanonicalPath() + File.pathSeparator +
```
Comment (LaCuneta) [inline src/HIDGogoExtension.java]: With the change for the `netLogoPackageExtras` setting, I think this should reference the copy in the root of the folder, so no `lib/`.

**Review by LaCuneta — CHANGES_REQUESTED** [review]
I understand about the difficulties in auto-detecting.  And good to know about the Linux native library.

The change for the packaging looks correct, just one small change in the code to match and I think we can merge.  Thanks again for all your work on this!

**Review by LaCuneta — APPROVED** [review]

**Pull request merged**